lib/Makefile.am

```diff
@@ -60,6 +66,29 @@ else
 endif
 endif
 
+$(top_builddir)/lib/%.ld: $(top_srcdir)/lib/%.ld.in
+	$(CPP) -xc -I$(top_srcdir)/include -C -P $< \
+	  | sed -n "/$$(sed -n '/^[^#]/{s/[*\/]/\\\0/g;p;q}' $<)/{:r;p;n;br}" \
+	  > $@
+
+# find the libqb.so symlink's target, if so, try to find out, iteratively,
+# its gradually shorter forms that likewise symlinks the same target as the
+# original libqb.so path, point to that file from the linker script using
+# qblog_script.ld as a template, and stored in place of original libqb.so
+# NOTE: readlink nor realpath are POSIX; not chained links ready
+install-exec-hook: $(top_builddir)/lib/qblog_script.ld
+	target=$$(ls -l "$(DESTDIR)$(libdir)/libqb.so" || :); \
+	  target=$${target#* -> }; t1_bn=$$(basename "$${target}" || :); \
+	  while test -n "$${t1_bn}"; do t2_bn=$${t1_bn%.*[0-9]*}; \
+	  test "$${t2_bn}" != libqb.so || break; \
+	  test -L "$${t2_bn}" || { t1_bn=$${t2_bn}; continue; }; \
+	  t2_target=$$(ls -l "$${t2_bn}" || break); t2_target=$${t2_target#* -> }; \
+	  test "$${t2_target}" = "$${target}" || break; \
+	  t1_bn=$${t2_bn}; done; test -n "$${t1_bn}" || \
+	  { echo "only applicable to SO symlink scheme"; exit 1; }; \
+	  echo "INPUT($${t1_bn})" > "$(DESTDIR)$(libdir)/libqb.so-t"; \
+	  cat $< >> "$(DESTDIR)$(libdir)/libqb.so-t"; \
+	  mv -f "$(DESTDIR)$(libdir)/libqb.so-t" "$(DESTDIR)$(libdir)/libqb.so"
 
```

Comment:
This looks horrible. Can it be tidied up or put in a script somewhere?

Reply:
Perhaps (look at libtool itself to see some really hard to read shell code).

But what's worse, 12 lines of well-wrapped (and overall commented code),
or having to distribute yet another single-use-only file?  I am biased towards
the latter :)

Reply:
Also, it gets a lot tidier when you observe `make install` screen :)

Reply:
I was actually able to separate last 2 lines, leaving the "blob" down to 10 :)